src/OpenTelemetry.AutoInstrumentation.Native/file_utils.h

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef OTEL_CLR_PROFILER_FILE_UTILS_H_
+#define OTEL_CLR_PROFILER_FILE_UTILS_H_
+
+#include <type_traits>
+
+#include "string_utils.h"
+
+#ifdef _WIN32
+#define PATH_TO_WSTRING(path_expr) (path_expr).wstring()
```

Comment:
Can we use std::filesystem::path functions instead: https://en.cppreference.com/w/cpp/filesystem/path/append.html